tests/components/elmax/test_config_flow.py

```diff
@@ -0,0 +1,268 @@
+"""Tests for the Abode config flow."""
+from unittest.mock import patch
+
+from elmax_api.exceptions import ElmaxBadLoginError, ElmaxBadPinError, ElmaxNetworkError
+
+from homeassistant import config_entries, data_entry_flow
+from homeassistant.components.elmax.const import (
+    CONF_ELMAX_PANEL_ID,
+    CONF_ELMAX_PANEL_NAME,
+    CONF_ELMAX_PANEL_PIN,
+    CONF_ELMAX_PASSWORD,
+    CONF_ELMAX_USERNAME,
+    DOMAIN,
+)
+from homeassistant.config_entries import SOURCE_REAUTH
+from homeassistant.data_entry_flow import FlowResult
+
+from tests.components.elmax import (
+    MOCK_PANEL_ID,
+    MOCK_PANEL_NAME,
+    MOCK_PANEL_PIN,
+    MOCK_PASSWORD,
+    MOCK_USERNAME,
+)
+
+CONF_POLLING = "polling"
+
+
+def _has_error(errors):
+    return errors is not None and len(errors.keys()) > 0
+
+
+async def _bootstrap(
+    hass,
+    source=config_entries.SOURCE_USER,
+    username=MOCK_USERNAME,
+    password=MOCK_PASSWORD,
+    panel_name=MOCK_PANEL_NAME,
+    panel_pin=MOCK_PANEL_PIN,
+) -> FlowResult:
+
+    result = await hass.config_entries.flow.async_init(
+        DOMAIN, context={"source": source}
+    )
+    if result["type"] != data_entry_flow.RESULT_TYPE_FORM or _has_error(
```

Comment:
Please don't include conditional logic in the test. Each test should just have one branch to make them easier to read and less complex and error prone.

Use pytest parametrization to reuse test logic instead.
https://docs.pytest.org/en/6.2.x/parametrize.html#pytest-mark-parametrize-parametrizing-test-functions

Reply:
Thanks @MartinHjelmare for the info. I believe pytest parametrization was not very useful for my needs. in any case I've refactored the tests to avoid conditional checks in #61273. It should now be more readable and maintainable as you suggest.